### PR TITLE
Add support for BPS and CS

### PR DESCRIPTION
**Pull request by johnwatson484**
https://dev.azure.com/defragovuk/DEFRA-FFC/_workitems/edit/158542
https://dev.azure.com/defragovuk/DEFRA-FFC/_workitems/edit/158543﻿

### CODE REVIEW

File: test/unit/processing/siti-agri/transform-invoice-lines.test.js [hunk 1]
```diff
@@ -1,11 +1,11 @@
 const transformInvoiceLine = require('../../../../app/processing/siti-agri/transform-invoice-line')
-const { sfi, sfiPilot, lumpSums } = require('../../../../app/schemes')
+const { sfi, sfiPilot, lumpSums, bps, cs } = require('../../../../app/schemes')
 
 describe('Transform invoice lines', () => {
   test('transforms SFI invoice line', async () => {
     const lineData = ['L', 'SFI0000001', '100', '2022', '80001', 'DRD10', 'SIP00000000001', 'RP00', 'N', '1', 'G00 - Gross value of claim', '2022-12-01', '2022-12-01', 'SOS273']
     const result = transformInvoiceLine(lineData, sfi.schemeId)
-    expect(result).toMatchObject({
+    expect(result).toEqual({
```
Comment (simonnineteen90) [inline test/unit/processing/siti-agri/transform-invoice-lines.test.js:8]: switched from `toMatchObject` to `toEqual` as it offers deep equality so obejcts have to match on all properties. `toMatchObject` was passing tests even if object was missing a property eg `correlationId`

Comment (Kirsten1822) [reply test/unit/processing/siti-agri/transform-invoice-lines.test.js:8]: I'm not too sure on this, so I'll let John have a look before resolving this comment

File: test/unit/processing/siti-agri/transform-invoice-lines.test.js [hunk 2]
```diff
@@ -51,6 +51,40 @@ describe('Transform invoice lines', () => {
...
+      agreementNumber: 'A01000000001/MT',
+      deliveryBody: 'NE00',
+      convergence: true,
+      description: 'G00 - Gross value of claim',
+      dueDate: '2023-12-01',
+      accountCode: 'SOS273'
+    })
+  })
+
+  test('transforms BPS invoice line', async () => {
```
Comment (Kirsten1822) [inline test/unit/processing/siti-agri/transform-invoice-lines.test.js:72]: This probably a nitpick, but could you move the bps test above cs? Just so it's consistent with the order of this 'const { sfi, sfiPilot, lumpSums, bps, cs } = require('../../../../app/schemes')'

File: test/unit/processing/siti-agri/transform-header.test.js [hunk 3]
```diff
@@ -48,6 +48,20 @@ describe('Transform header', () => {
     })
   })
 
+  test('for SFI return undefined if paymentRequestNumber is NaN', async () => {
```
Comment (Kirsten1822) [inline test/unit/processing/siti-agri/transform-header.test.js:51]: I wonder if these tests for SFI should be moved above SFI Pilot?

**Review by simonnineteen90 — COMMENTED** [review]
Happyto approve all changes made by @johnwatson484 and @Kirsten1822

File: test/unit/processing/siti-agri/transform-invoice-lines.test.js (hunk 1 above)
Comment (simonnineteen90) [reply test/unit/processing/siti-agri/transform-invoice-lines.test.js:8]: Good call, im not 100% on it but it made sense as it made the test a bit stricter but happy to wait and see what John thinks.

File: test/unit/processing/siti-agri/transform-invoice-lines.test.js (hunk 2 above)
Comment (simonnineteen90) [reply test/unit/processing/siti-agri/transform-invoice-lines.test.js]: agree, changes made.

File: test/unit/processing/siti-agri/transform-header.test.js (hunk 3 above)
Comment (simonnineteen90) [reply test/unit/processing/siti-agri/transform-header.test.js]: agree, updated.

File: test/unit/processing/siti-agri/transform-invoice-lines.test.js (hunk 1 above)
Comment (johnwatson484) [reply test/unit/processing/siti-agri/transform-invoice-lines.test.js:8]: I'd agree it should be `toEqual` if you want to ensure all properties match.

`toMatchObject` is useful you want to make sure that properties of an object are included along with anything else that may be outside the object.

For really complicated objects you may also need `toStrictEqual` but not in this case.

**Review by Kirsten1822 — APPROVED** [review]
Happy to approve :)

**Pull request merged**